src/entities/protocols/seaport.ts

```diff
@@ -1,19 +1,24 @@
-import abi from '../../../abis/Seaport.json'
 import { BigNumber, BigNumberish } from 'ethers'
 import { Interface } from '@ethersproject/abi'
+import abi from '../../../abis/Seaport.json'
 import { BuyItem, Market, NFTTrade, TokenType } from '../NFTTrade'
 import { TradeConfig } from '../Command'
 import { RoutePlanner, CommandType } from '../../utils/routerCommands'
-
-export enum SeaportVersion {
-  V1_1,
-  V1_4,
-}
+import { encodeHandleInputTokens, Permit2Permit } from '../../utils/inputTokens'
+import { ETH_ADDRESS } from '../../utils/constants'
 
 export type SeaportData = {
   items: Order[]
   recipient: string // address
-  version: SeaportVersion
+  protocolAddress: string
+  inputTokenProcessing?: InputTokenProcessing[]
+}
+
+export type InputTokenProcessing = {
+  token: string
+  permit2Permit?: Permit2Permit
+  protocolApproval: boolean
+  permit2TransferFrom: boolean
```

Comment:
nit: place `protocolApproval` above `permit2Permit` to follow order of arguments in `encodeHandleInputTokens`